### PR TITLE
Add support for graphql pagination with episodes & aka titles

### PR DESCRIPTION
**Pull request by mhdzumair**

### CODE REVIEW

**Comment by mhdzumair** [comment]
This was a long time feature that I was added. But not make pr. now i created it. 
seems like ruff is failing. i would like to set max line length as 120

**Comment by mhdzumair** [comment]
Fix #7

**Comment by mhdzumair** [comment]
Hi sorry for putting everything in this PR. i need some changes in my fork repo. you can review this PR cherry-pick what best for your release.

**Comment by uyar** [comment]
Hi. Thanks for your work. Yes, I'll need to do some cherry picking because I had already fixed some of the tests.

File: pyproject.toml
Comment (uyar) [inline pyproject.toml:1]: I don't like long lines. I prefer to keep the rule at 79 and put an explicit `# noqa: E501` when I can't avoid it.

Comment (mhdzumair) [reply pyproject.toml:1]: done

**Comment by mhdzumair** [comment]
@uyar I need to ask, any reason why you stick to URL for fetching the data?

**Comment by uyar** [comment]
I don't want to encourage flooding the IMDb servers with this package. I also don't want to have many external dependencies. But there are three of them already and I guess it's OK to use httpx. If you prepare a PR for this, I am very likely to approve it. But please wait until I'm doing merging this one. And keep the PR limited to httpx migration only.

**Comment by mhdzumair** [comment]
sure, please do the merge so that i can create other PRS. i have completed the search title function as well as migration to httpx. almost completed the paginated search with graphql endpoint. im in hurry to push everything. xD 😅😅

**Comment by uyar** [comment]
I'm leaving out the proxy related code for now. Can you do this via httpx? Or maybe it could be handled through environment variables. I'd prefer not to have socks-specific code.

**Comment by mhdzumair** [comment]
Merge this mr, i will remove that proxy related config as well as socks code in httpx mr.

**Comment by uyar** [comment]
Really great work, thanks a lot! I've merged most of it. Some notes:

- I did not merge SOCKS related code, sorry :( I don't want that dependency even temporarily at the moment.
- Please don't bundle irrelevant changes into the same commits and PRs. It makes it very difficult to review.
- Please don't make changes to the project settings without asking. Like changing ruff rules.
- Make sure that your files end in a newline.
- If possible, also make sure that your JSON spec files are properly indented.

I couldn't fix the parental guide tests. You might want to take another look at them.

Again, thanks for the great work! I've added your name to the authors metadata rather than the contributors file.

**Pull request closed without merging**